chips/sam4l/src/spi.rs

```diff
@@ -172,54 +213,52 @@ impl Spi {
     }
 
     fn init_as_role(&self, role: SpiRole) {
-        let regs: &SpiRegisters = unsafe { &*self.registers };
+        let regs: &Registers = unsafe { &*self.registers };
 
         self.role.set(role);
         self.enable_clock();
 
         if self.role.get() == SpiRole::SpiMaster {
             // Only need to set LASTXFER if we are master
-            regs.cr.set(spi_consts::cr::LASTXFER);
+            regs.cr.write(Control::LASTXFER::SET);
         }
 
         // Sets bits per transfer to 8
         let mut csr = self.read_active_csr();
-        csr &= !spi_consts::csr::BITS_MASK;
-        csr |= spi_consts::csr::BITS8;
+        csr &= !(0x1111 << 4);
+        csr |= 0b0000 << 4;
```

Comment:
Why aren't you using the fancy flags to set the value here?

Reply:
Any ideas how to? The SPI is weird because there are multiple registers for the different chip selects, and I don't know how to use the new setup indirectly.

Reply:
I've got a solution here I'm writing a commit for. Actually I think nothing needs to change in the way regs work, and SPI was just doing something that doesn't make sense with the new regs stuff.